search-app/src/cmr/search/services/query_service.clj

```diff
@@ -301,6 +304,102 @@
+   collection concept ids in the given result format."
+  [context coll-concept-ids result-format]
+  (when (seq coll-concept-ids)
+    ;; find all collection revisions, then filter out the highest revisions
+    ;; and replace the hits and items in results with those of the highest revisions.
+    (let [condition (gc/and-conds
+                     [(qm/string-conditions :concept-id coll-concept-ids true)
+                      (qm/boolean-condition :deleted false)])
+          query (qm/query {:concept-type :collection
+                           :condition condition
+                           :all-revisions? true
+                           :page-size :unlimited
+                           :result-format result-format})
+          results (qe/execute-query context query)
+          highest-coll-revisions (u/map-values
+                                  #(apply max (map :revision-id %))
+                                  (group-by :concept-id (:items results)))
+          highest-revisions (filter
+                             #((set highest-coll-revisions)
+                               [(:concept-id %) (:revision-id %)])
+                             (:items results))]
+      (-> results
+          (assoc :items highest-revisions)
+          (assoc :hits (count highest-revisions))))))
+
+(defn get-deleted-collections
+  "Executes elasticsearch searches to find collections that are deleted.
+   This only finds collections that are truely deleted and not searchable.
+   Collections that are deleted, then later ingested again are not included in the result.
+   Returns references to the highest collection revision prior to the collection tombstone."
+  [context params]
+  (pv/validate-deleted-collections-params params)
+  ;; 1/ Find all collection revisions that are deleted after satify the revision-date query -> c1
+  ;; 2/ Filters out any collections c1 that still exists -> c2
+  ;; 3/ Find all collection revisions for the c2, return the highest revisions that are visible
+  (let [start-time (System/currentTimeMillis)
+        result-format (:result-format params)
+        coll-concept-ids (get-collections-with-deleted-revisions context params)
+        visible-concept-ids (get-visible-collections context coll-concept-ids)
+        ;; Find the concept ids that are still deleted
+        deleted-concept-ids (seq (set/difference
+                                   (set coll-concept-ids)
+                                   (set visible-concept-ids)))
+        results (get-highest-visible-revisions context deleted-concept-ids result-format)
+        ;; when results is nil, hits is 0
+        results (or results {:hits 0 :items []})
+        total-took (- (System/currentTimeMillis) start-time)
+        ;; construt the response results string
+        results-str (common-search/search-results->response
+                     context
+                     ;; pass in a fake query to get the desired response format
```

Comment:
The `search-results->response` function takes a query argument and use it to determine the response body that it will generate. This fake query is to set up the fields that will be used to generate the correct response.